### ISSUE TITLE
>hrl files present in the include directory of project are not recognized in the src directory of the project 

### ISSUE THREAD

**Issue by blacklightknight** [erlang/erlide_eclipse#298]
I was studying the source code of ejabberd on eclipse. i figured out that each of my .erl files present the src folder of the project with a include statement in it was showing an compilation error on saving the project.
When I searched for the errors it said that it was not able to include the .hrl files present in the include folder.
I wanted to know that why is this happening as the files placed in include directory must be recognized in the src folder of the same project.
i have also tried the solutions given on this link 
http://www2.erlangcentral.org/wiki/?title=Compile_With_Hard_To_Find_Include_Files_In_Erlang_Shell
but this also didn't worked for me.
Anyone who has any Idea about the issue.

**Comment by vladdu** [comment]
I imported the project, without building/compiling anything (can't on this machine). The only include files that couldn't be found are:
- XmppAddr.hrl 
- xmpp.hrl
- fxml.hrl
- jid.hrl
- ns.hrl
which don't exist and I assume are being generated by the build (for example the first one from asn1). 

If you look for those files and see what directories they are located in, then you can change the eclipse project's configuration (right-click->properties->Erlang) and add those directories to 'include'. Then it should work.

Please note that parse transforms aren't handled and the errors related to that will remain.

**Comment by vladdu** [comment]
Maybe I should have asked first: how did you create the ejabberd project in eclipse? Is it configured properly (src and include directories)?

**Comment by blacklightknight** [comment]
Vlad Dumitrescu, first of all let me thank you a lot for replying to my problem.I am new to ejabberd and erlang and it has been very tough to get any reply.So thank you really.

I would like to tell you that I have solved the issue.The problem was that I was running my code on eclipse and in my project structure all my .hrl files were placed inside include directory.And in eclipse if this happens then you have to give the complete path with double slashes in it to be able to recognize the .hrl files. Or the second option is to place all the .hrl files in the src directory only right from the start itself.

Thank you anyway.

**Comment by vladdu** [comment]
If your Eclipse project is properly configured as an Erlang project, then the 'include' directory is recognized. Do your project's properties look like below? All source directories and all include directories should be listed, if there are more than one. 

![untitled](https://cloud.githubusercontent.com/assets/9707/23498722/3a64a398-ff29-11e6-957b-979a8636aa43.png)

**Comment by blacklightknight** [comment]
![scr](https://cloud.githubusercontent.com/assets/26063163/23499099/bfe0a28e-ff50-11e6-93ff-fb3aa4973846.png)
![scr2](https://cloud.githubusercontent.com/assets/26063163/23499106/c68db43c-ff50-11e6-97ec-afe9b83599ac.png)
yes Vlad my Eclipse project is properly configured as an Erlang project.

**Comment by vladdu** [comment]
It should have found the hrl files in include too. Is this your own project, or ejabberd?

**Comment by blacklightknight** [comment]
I first create an erlang project then I import the source code of ejabberd into it.

**Comment by vladdu** [comment]
Try this: Build first ejabberd from the command line. Then create a new project in Eclipse and in the "location" field enter the location of the ejabberd code. Then on the "project layout" page, press "discover paths". All directories including .erl or .hrl files will be added automatically. 

This should give you better configuration that hopefully finds all files.

**Comment by blacklightknight** [comment]
Actually, I am using windows platform so I had already built ejabberd using the windows installer.I didn't get the meaning of "location","project layout", and "discover paths" you mentioned.

**Comment by vladdu** [comment]
When creating a project, you get to set the name and there is a field for specifying the location. Uncheck the "use default" and browse to wherever the ejabberd code is.

Then on the third page (the second image you attached) you have the project layout configuration. Since there is already code, the button "Discover paths..." should be enabled, press it to fill the paths automatically.

**Comment by blacklightknight** [comment]
Yes I did this but still .hrl files are not recognized from src folder.you can see:
![scr3](https://cloud.githubusercontent.com/assets/26063163/23502591/5bdc87b8-ff5e-11e6-8b2e-da8562dadb6e.png)

**Comment by vladdu** [comment]
Oh, I see. This is not how erlide works. You don't need to compile from the console, but erlide is doing it for you. If you set `Project->Build automatically` enabled, you only have to check the `Problems` view for any warnings or errors. When you save a change, that file will be recompiled automatically.

**Comment by blacklightknight** [comment]
yes I knew that way too, but the problem still continues this way as yo can see in the following screenshots:
![a](https://cloud.githubusercontent.com/assets/26063163/23599831/61732ebc-0269-11e7-9f59-11b72784b5ea.png)
![b](https://cloud.githubusercontent.com/assets/26063163/23599837/6749eeb6-0269-11e7-993d-be85c9ebdd73.png)

**Comment by vladdu** [comment]
Do I see right that the Erlang file has no syntax highlighting? Then it's possible that there's something wrong with the communication with the backend. 

Anyway, did you build ejabberd before importing the project? The jid.hrl file, for example, is part of xmpp, which is fetched in ./deps when runing make. Then when creating the eclipse project and pressing "discover paths", you will get all dependencies. The full suport for rebar projects is in the works.

There will probably still be errors, mostly because of the lager parse transform, which is not well supported by erlide. It doesn't know that it should compile the parse transform first. Usually, doing a clean and rebuild (from eclipse) might help.

**Comment by blacklightknight** [comment]
Yes I already did build ejabberd as you can see here I have got the dependency folder:
![abc](https://cloud.githubusercontent.com/assets/26063163/23643777/bda9c234-0329-11e7-8ee7-2ce1e81729b9.png)

**Comment by vladdu** [comment]
And you created the eclipse project after that, using the "discover paths" button? In the project properties you should see all the `deps/.../src` and `deps/.../include` directories.

**Comment by blacklightknight** [comment]
yes exactly, but still the problem continues.

**Comment by vladdu** [comment]
Very strange! It works for me... Do you have "Project->Build automatically" enabled? Did you try "Project->Clean"? 

Please go to Window->Preferences->Erlang->Report problems, create a report and send it to me (or attach it here). Maybe I can see something there.

**Comment by blacklightknight** [comment]
Yes my "Project->Build automatically" is enabled and I did use "Project->Clean"

**Comment by vladdu** [comment]
Please attach an problem report like described above.
